spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskExecutionController.java

```diff
@@ -198,10 +203,20 @@ public TaskExecutionResource instantiateResource(TaskJobExecutionRel taskJobExec
 	private Page<TaskJobExecutionRel> getPageableRelationships(Page<TaskExecution> taskExecutions, Pageable pageable){
 		List<TaskJobExecutionRel> taskJobExecutionRels = new ArrayList<>();
 		for(TaskExecution taskExecution: taskExecutions.getContent()) {
-			taskJobExecutionRels.add( new TaskJobExecutionRel(taskExecution,
-					new ArrayList<>(explorer.getJobExecutionIdsByTaskExecutionId(
+			taskJobExecutionRels.add( new TaskJobExecutionRel(sanitizePotentialSensitiveKeys(taskExecution),
+					new ArrayList<>(this.explorer.getJobExecutionIdsByTaskExecutionId(
 							taskExecution.getExecutionId()))));
 		}
 		return new PageImpl<>(taskJobExecutionRels,pageable,taskExecutions.getTotalElements());
 	}
+
+	private TaskExecution sanitizePotentialSensitiveKeys(TaskExecution taskExecution) {
+		ArgumentSanitizer argumentSanitizer = new ArgumentSanitizer();
```

Comment:
should we create a new one each invocation? could we reuse one created for this instance?